### PR TITLE
AppleWin Remote Control

### PR DESCRIPTION
**Pull request by hasseily**
This pull request provides Applewin with a remote control framework.
It also adds parsing of some metadata for WOZ files (title, subtitle, version) and HDV files (Volume string).

Also included:

Documentation
Command line parameters
Preferences window
Background sound fix

### CODE REVIEW

**Comment by tomcw** [comment]
Hi Henri,

Thanks for submitting this PR, which relates (in part) to feature #848.

I have been discussing this offline with other AppleWin maintainers. Our conclusion is that this work is of questionable value and currently not a good fit for inclusion in mainline AppleWin, as:

a) it's specifically for a commercial product (Grid Cartographer)
b) it's highly niche, and of little benefit to mainstream AppleWin users
c) for me, I'd want a more generalised solution, that includes interfacing with the built-in debugger

In terms of the PR itself, and without digging too much into the details of your evolving implementation, there are many areas that we'd want re-written/re-designed before we could think about accepting this work. Please see [CONTRIBUTING.md](https://github.com/AppleWin/AppleWin/blob/master/CONTRIBUTING.md) file for some high-level pointers.

We encourage you to continue your development of this feature out-of-tree in your own fork. This doesn't preclude us from having an interest in this work in the future, but right now we don't feel it is a good fit.

**Comment by hasseily** [comment]
Well understood. Thanks for the feedback. I'll try some AI play with AppleWin and I'll also work on the Woz metadata parsing.

**Review by Michaelangel007 — COMMENTED** [review]
Comments don't need a `RIK` tag -- diff tools will already show us the commit and author of changes.

File: source/Memory.cpp [hunk 1]
```diff
@@ -1816,11 +1834,11 @@ inline DWORD getRandomTime()
 void MemReset()
 {
 	// INITIALIZE THE PAGING TABLES
-	ZeroMemory(memshadow,256*sizeof(LPBYTE));
-	ZeroMemory(memwrite ,256*sizeof(LPBYTE));
+	ZeroMemory(memshadow, 256 * sizeof(LPBYTE));
+	ZeroMemory(memwrite, 256 * sizeof(LPBYTE));
```
[the remark this answers is not in the document]
Comment (hasseily) [reply source/Memory.cpp:1838]: That's VS autoformatting. Will keep it in mind, thanks.

File: source/Memory.cpp [hunk 2]
```diff
@@ -1842,6 +1860,9 @@ void MemReset()
 	// OR
 	//   F2, Ctrl-F2, F7, HGR
 	DWORD randTime = getRandomTime();
+
+	g_nMemoryClearType = MIP_FF_FF_00_00;
```
[the remark this answers is not in the document]
Comment (hasseily) [reply source/Memory.cpp:1864]: Yes my mistake. I had set it to something non-random for some other work and then forgot to delete it. It is initialized on line 111:
`int			g_nMemoryClearType = MIP_FF_FF_00_00; // Note: -1 = random MIP in Memory.cpp MemReset()`
and should never be set again unless through command-line

File: source/Applewin.cpp [hunk 3]
```diff
@@ -1876,6 +1897,10 @@ static bool ProcessCmdLine(LPSTR lpCmdLine)
 			g_cmdLine.strCurrentDir = lpCmdLine;
 		}
 		else if (strcmp(lpCmdLine, "-no-nsc") == 0)
+		{
```
[the remark this answers is not in the document]
Comment (hasseily) [reply source/Applewin.cpp:1900]: Absolutely

**Pull request closed without merging**